base_layer/core/src/proof_of_work/diff_adj_manager/diff_adj_storage.rs

```diff
@@ -112,6 +115,19 @@ where T: BlockchainBackend
         })
     }
 
+    /// Returns the estimated target difficulty for the specified PoW algorithm.
+    pub fn get_median_timestamp(&mut self) -> Result<EpochTime, DiffAdjManagerError> {
+        self.update()?;
+        let mut length = self.timestamps.len();
+        if length == 0 {
+            return Err(DiffAdjManagerError::EmptyBlockchain);
+        }
+        let mut sorted_timestamps: Vec<EpochTime> = self.timestamps.clone().into();
+        sorted_timestamps.sort();
+        length = (length / 2); // we want the median, should be index 5 (11/2)
+        Ok(sorted_timestamps[length])
```

Comment:
When there is less than 11 blocks, or the constant is configured differently, it might happen that you have an even number of timestamp samples. The median of an even number of elements is the average of the center 2 values.

Reply:
I believe this adds unnecessary complexity. Choosing the number is a consensus thing. We should just avoid choosing an even number. Even choosing an even number is fine as long as the median timestamp is then defined as such then.  
This code is only here so that the median timestamp works for the first 11 blocks of the chain.

Reply:
I was kinda expecting to see a 
```
pub fn median<T: PartialOrd>(data: &[T]) -> T {
...
}
```

function that calculates the median Of anything it receives, and calling thru to that

Easy to unit test and flexible. Next PR

Reply:
Median has a very precise mathematical definition though. You're not wrong that we can make consensus rules whatever, but it's not right to call this median then.